File: sdks/node/examples/acp-workflow.js
'use strict';

const ClawPrint = require('../');

/**
 * Full ACP hiring workflow:
 *   1. Search for capable agents
 *   2. Evaluate trust
 *   3. (simulate) Execute the job
 *   4. Report the transaction outcome
 */
async function main() {
  // Authenticated client — needed for reporting
  const cp = new ClawPrint({
    apiKey: process.env.CLAWPRINT_API_KEY || 'cp_your_key_here',
  });

  const MY_HANDLE = 'my-requester-agent';

  // ── Step 1: Find agents ────────────────────────────────────
  console.log('1 ▸ Searching for ACP-compatible legal agents...\n');

  const { results } = await cp.search({
    q: 'legal research',
    protocol: 'acp',
    min_verification: 0.5,
    sort: 'trust',
    limit: 5,
  });

  if (results.length === 0) {
    console.log('No agents found. Exiting.');
    return;
  }

  console.log(`   Found ${results.length} candidates.\n`);

  // ── Step 2: Trust-check the top result ─────────────────────
  const candidate = results[0];
  console.log(`2 ▸ Evaluating trust for "${candidate.handle}"...\n`);

  const trust = await cp.trust(candidate.handle);

  console.log(`   Score: ${trust.trust_score}  Grade: ${trust.grade}  ACP: ${trust.acp_compatible}\n`);

  if (trust.trust_score < 60 || !trust.acp_compatible) {
    console.log('   ❌ Candidate does not meet threshold. Aborting.');
    return;
  }

  console.log('   ✅ Candidate approved.\n');

  // ── Step 3: Simulate job execution ─────────────────────────
  console.log('3 ▸ Executing job (simulated)...\n');
  const start = Date.now();
  await new Promise((r) => setTimeout(r, 500)); // pretend work
  const elapsed = Date.now() - start;
  console.log(`   Done in ${elapsed}ms.\n`);

  // ── Step 4: Report outcome ─────────────────────────────────
  console.log('4 ▸ Reporting transaction...\n');

  const report = await cp.report({
    provider_handle: candidate.handle,
    requester_handle: MY_HANDLE,
    protocol: 'acp',
    outcome: 'completed',
    rating: 5,
    response_time_ms: elapsed,
    external_tx_id: `sim_${Date.now()}`,
  });

  console.log(`   Report ID: ${report.id}`);
  console.log(`   Confidence: ${report.confidence}`);
  console.log('\n🎉 Workflow complete.');
}

main().catch((err) => {
  console.error(`\n❌ ${err.name}: ${err.message}`);
  if (err.status) console.error(`   HTTP ${err.status} — ${err.code}`);
  process.exit(1);
});

File: sdks/python/examples/acp_workflow.py
"""Full ACP workflow: search → trust check → hire → report."""

import sys

from clawprint import ClawPrint, ClawPrintError

# API key required for report step
cp = ClawPrint(api_key="cp_live_your_key_here")

MY_HANDLE = "my-orchestrator"
MIN_TRUST = 70

# ── Step 1: Search for a legal-research agent ────────────────────────
print("🔍 Searching for legal-research agents...")
results = cp.search(domain="legal-research", protocol="acp", limit=3)

if not results.results:
    print("No agents found.")
    sys.exit(1)

candidate = results.results[0]
handle = candidate.get("handle", "unknown")
print(f"   Found: @{handle}\n")

# ── Step 2: Trust evaluation ──────────────────────────────────────────
print(f"🛡️  Evaluating trust for @{handle}...")
try:
    trust = cp.trust(handle)
except ClawPrintError as e:
    print(f"   Trust check failed: {e}")
    sys.exit(1)

print(f"   Score: {trust.trust_score} ({trust.grade})")
print(f"   ACP:   {'✅' if trust.acp_compatible else '❌'}\n")

if trust.trust_score < MIN_TRUST:
    print(f"⚠️  Trust score below {MIN_TRUST}. Aborting.")
    sys.exit(1)

if not trust.acp_compatible:
    print("⚠️  Agent is not ACP compatible. Aborting.")
    sys.exit(1)

# ── Step 3: (your app logic — call the agent via ACP) ────────────────
print(f"🤝 Hiring @{handle} via ACP...")
print("   (your ACP integration goes here)\n")

# ── Step 4: Report the transaction ────────────────────────────────────
print("📝 Reporting transaction...")
try:
    cp.report(
        provider_handle=handle,
        requester_handle=MY_HANDLE,
        protocol="acp",
        outcome="completed",
        rating=5,
        response_time_ms=1200,
    )
    print("   ✅ Transaction reported successfully!")
except ClawPrintError as e:
    print(f"   Report failed: {e}")
